armi/runLog.py

```diff
@@ -446,18 +446,22 @@ def filter(self, record):
         # determine if this is a "do not duplicate" message
         msg = str(record.msg)
         single = getattr(record, "single", False)
-        label = getattr(record, "label", msg)
-        label = msg if label is None else label
 
         # If the message is set to "do not duplicate" we may filter it out
         if single:
             if record.levelno in (logging.WARNING, logging.CRITICAL):
+                # grab the custom label, if it exists, for warning logging later
+                label = getattr(record, "label", msg)
+                label = msg if label is None else label
```

Comment:
Don't these two lines duplicate each other? Because you default to `msg` from `getattr()`

Reply:
Well, the default `label` is `None`.  So I think `getattr(record, "label", msg)` returns `None` most of the time.

I could try it and find out though.

Reply:
If the default label is `None`, wouldn't `getattr(record, "label", msg)` return `msg` instead of `None`?

Reply:
@keckler @opotowsky 

The issue is this:

```python
>>> class X:
...     def __init__(self, x):
...         self.label = None
...         self.x = x
...
>>> x = X(7)
>>>
>>> getattr(x, "x", "hi")
7
>>> getattr(x, "label", "msg")
None
```

When `x.label` exists, but is set to the default value `None`, the `getattr()` correctly finds it and returns `None` as the value.

Reply:
I have attempted to clarify the situation with better comments:

```python
# if this is from the custom logger, it will have a "label"
label = getattr(record, "label", msg)
# the "label" default is None, which needs to be replaced
label = msg if label is None else label
```